server/src/main/java/com/flipkart/varadhi/events/EntityStateProcessor.java

```diff
@@ -0,0 +1,270 @@
+package com.flipkart.varadhi.events;
+
+import com.flipkart.varadhi.cluster.messages.ClusterMessage;
+import com.flipkart.varadhi.cluster.messages.ResponseMessage;
+import com.flipkart.varadhi.common.VaradhiCache;
+import com.flipkart.varadhi.common.events.EntityEvent;
+import com.flipkart.varadhi.common.events.EventType;
+import com.flipkart.varadhi.entities.InternalCompositeTopic;
+import com.flipkart.varadhi.entities.Project;
+import com.flipkart.varadhi.entities.StorageTopic;
+import com.flipkart.varadhi.entities.VaradhiTopic;
+import com.flipkart.varadhi.entities.auth.ResourceType;
+import com.flipkart.varadhi.produce.services.ProducerService;
+import com.flipkart.varadhi.services.ProjectService;
+import com.flipkart.varadhi.spi.services.Producer;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.EnumMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.BiConsumer;
+import java.util.function.Function;
+
+/**
+ * Processes entity state changes and updates in-memory state accordingly.
+ * <p>
+ * This processor uses a functional approach to manage entity state updates for different resource types,
+ * providing a clean and extensible way to handle entity events across the system.
+ * <p>
+ * The processor directly handles cluster messages containing entity events, extracting the events
+ * and processing them appropriately.
+ */
+@Slf4j
+public final class EntityStateProcessor {
```

Comment:
it is a dispatcher. it should only contain dispatch logic.

project, topic related code probbaly should go into their own service class.